### PR TITLE
fifechan: link with sdl2 not sdl

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----
Current `bottle` doesn't have `libfifechan_sdl.0.1.4.dylib`. 
```
$ brew fetch fifechan
==> Downloading https://homebrew.bintray.com/bottles/fifechan-0.1.4.high_sierra.
######################################################################## 100.0%
Downloaded to: /Users/s172262/Library/Caches/Homebrew/fifechan-0.1.4.high_sierra.bottle.tar.gz
SHA256: 5eba713dbc0b32b14703d9423c61b4272984eb6c6d24f10cb2f7fbca30ec4581

$ tar xf fifechan-0.1.4.high_sierra.bottle.tar.gz
$ cd fifechan/0.1.4/lib/
$ ls
libfifechan.0.1.4.dylib		libfifechan_opengl.0.1.4.dylib
libfifechan.dylib		libfifechan_opengl.dylib

$ otool -L lib*.dylib |sed -e "/:/d" |sort |uniq
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	@@HOMEBREW_PREFIX@@/opt/fifechan/lib/libfifechan.0.1.4.dylib (compatibility version 0.1.4, current version 0.1.4)
	@@HOMEBREW_PREFIX@@/opt/fifechan/lib/libfifechan_opengl.0.1.4.dylib (compatibility version 0.1.4, current version 0.1.4)
	@loader_path/libfifechan.0.1.4.dylib (compatibility version 0.1.4, current version 0.1.4)
```
This commite builds `libfifechan_sdl.0.1.4.dylib`.
```
$ brew list fifechan
/usr/local/Cellar/fifechan/0.1.4/include/fifechan/ (71 files)
/usr/local/Cellar/fifechan/0.1.4/include/fifechan.hpp
/usr/local/Cellar/fifechan/0.1.4/lib/libfifechan.0.1.4.dylib
/usr/local/Cellar/fifechan/0.1.4/lib/libfifechan_opengl.0.1.4.dylib
/usr/local/Cellar/fifechan/0.1.4/lib/libfifechan_sdl.0.1.4.dylib
/usr/local/Cellar/fifechan/0.1.4/lib/ (3 other files)

 $ otool -L /usr/local/Cellar/fifechan/0.1.4/lib/libfifechan_sdl.0.1.4.dylib
/usr/local/Cellar/fifechan/0.1.4/lib/libfifechan_sdl.0.1.4.dylib:
	/usr/local/opt/fifechan/lib/libfifechan_sdl.0.1.4.dylib (compatibility version 0.1.4, current version 0.1.4)
	/usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/local/opt/sdl2_image/lib/libSDL2_image-2.0.0.dylib (compatibility version 1.0.0, current version 1.1.0)
	@loader_path/libfifechan.0.1.4.dylib (compatibility version 0.1.4, current version 0.1.4)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

### CODE REVIEW

**Comment by DomT4** [comment]
Have no opinions on the actual PR but as a rule of thumb if you're changing default/recommended dependencies the formula's `revision` should get bumped too.

**Comment by s172262** [comment]
So none notices this...
```
$ brew uses fifechan --include-optional
fifengine
```

File: Formula/fifechan.rb [hunk 1]
```diff
@@ -14,12 +15,15 @@ class Fifechan < Formula
 
   depends_on "cmake" => :build
   depends_on "allegro" => :recommended
-  depends_on "sdl" => :recommended
-  depends_on "sdl_image" => :recommended
+  depends_on "sdl2" => :recommended
```
Comment (ilovezfs) [inline Formula/fifechan.rb]: let's just get rid of all of the `=> :recomended` and make them required. no one has been using he without options anyway

File: Formula/fifechan.rb [hunk 2]
```diff
@@ -13,13 +14,16 @@ class Fifechan < Formula
...
+  depends_on "allegro"
+  depends_on "sdl2"
+  depends_on "sdl2_image"
+  depends_on "sdl2_ttf"
 
   def install
     mkdir "build" do
-      system "cmake", "..", *std_cmake_args
+      args = std_cmake_args
+      args << "-DENABLE_SDL_CONTRIB=ON"
```
Comment (ilovezfs) [inline Formula/fifechan.rb]: You can just inline this below.

```
       system "cmake", "..", "-DENABLE_SDL_CONTRIB=ON", *std_cmake_args
```

**Comment by ilovezfs** [comment]
Thanks again @s172262!

**Pull request closed without merging**